core/src/main/java/com/scalar/db/transaction/consensuscommit/TwoPhaseConsensusCommit.java

```diff
@@ -203,6 +208,7 @@ public void commit() throws CommitException, UnknownTransactionStatusException {
 
       commit.commitRecords(crud.getSnapshot());
       status = Status.COMMITTED;
+      removeActiveTransaction(getId());
```

Comment:
Is there any particular reason for not doing `manager.removeActiveTransaction(getId())` and adding `removeActiveTransaction` to the `AbstractTwoPhaseConsensusCommit`?

Reply:
Honestly, I don't have any strong reasons for that, and I think either is fine. But, we need to make the scope of the field `transactionManager` of `AbstractTwoPhaseConsensusCommit` `protected` in that case, which I don't really like actually. I feel using a field of a superclass in a subclass is more like tightly-coupled and not explicit. So I don't do that very often.

That's why I use the method (`removeActiveTransaction`) of `AbstractTwoPhaseConsensusCommit` instead of the field `transactionManager` directly. Another advantage of this is that we can prevent exposing unnecessary methods of `transactionManager` to subclasses unintendedly.

Reply:
@feeblefakie As discussed, I applied decorator pattern to mange active transactions. Please take a look again when you have time!

Reply:
@brfrn169 Looks pretty good! Thank you!